tests/internal/remoteconfig/test_remoteconfig.py

```diff
@@ -398,3 +398,33 @@ def test_remote_configuration_check_remote_config_enable_in_agent_errors(
     assert worker._state == worker._online if expected else worker._agent_check
     worker.stop_subscribers(True)
     worker.disable()
+
+
+@pytest.mark.subprocess(
+    parametrize=dict(
+        DD_REMOTE_CONFIGURATION_ENABLED=["1", "0"],
+    ),
+)
+def test_rc_default_products_registered():
+    """
+    By default, RC should be enabled. When RC is enabled, we will always
+    enable the tracer flare feature as well. There should be three products
+    registered when DD_REMOTE_CONFIGURATION_ENABLED is True
+    """
+    import os
+
+    from ddtrace.internal.utils.formats import asbool
+
+    rc_enabled = asbool(os.environ.get("DD_REMOTE_CONFIGURATION_ENABLED"))
+
+    # Import this to trigger the preload
+    from ddtrace import config
+    import ddtrace.auto  # noqa:F401
+
+    assert config._remote_config_enabled == rc_enabled
+
+    from ddtrace.internal.remoteconfig.worker import remoteconfig_poller
+
+    assert bool(remoteconfig_poller._client._products.get("APM_TRACING")) == rc_enabled
```

Comment:
Yeah it turns out if I had dug a bit more, I would've realized that my environment variables weren't getting set at all. So the reason I was getting 0 products wasn't because I wasn't accessing the correct RC poller, but because it was correctly not registering anything because `DD_REMOTE_CONFIGURATION_ENABLED` is set to False by default in the riotfile.py. The agent also doesn't need to be available, and it just sends a debug log if it can't reach it for any reason (I think this makes sense in case your app is up, but the agent is taking a bit to come up, then that shouldn't be an issue since you can keep polling to see if it eventually comes up).